### PR TITLE
Update CFN with DLAMI Conda v16.0

### PR DESCRIPTION
**Pull request by mirocody**
1. Update DLAMI ami-ids-- to DLAMI Conda V16
2. Update README
3. Update mxnet submodule

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,34 +93,39 @@ Follow **[Step 3](cfn-template/StackSetup.md#logintomaster)** in [Using the AWS
...
-    git submodule update --init $EFS_MOUNT/deeplearning-cfn/examples/mxnet && \
-    cd $EFS_MOUNT/deeplearning-cfn/examples/mxnet/ && \
-    git submodule update --init $EFS_MOUNT/deeplearning-cfn/examples/mxnet/dmlc-core
+    git submodule update --init $EFS_MOUNT/deeplearning-cfn/examples/incubator-mxnet && \
+    cd $EFS_MOUNT/deeplearning-cfn/examples/incubator-mxnet && \
+    git submodule update --init $EFS_MOUNT/deeplearning-cfn/examples/incubator-mxnet/3rdparty/dmlc-core
+	# Also we need make a minor change to dmlc-core to use bash activate conda environment remotely
+	vi $EFS_MOUNT/deeplearning-cfn/examples/incubator-mxnet/3rdparty/dmlc-core/tracker/dmlc_tracker/tracker.py
+	#Then goto line 366 change code from
+	target=(lambda: subprocess.check_call(self.cmd, env=env, shell=True)), args=())
```
Comment (nswamy) [inline README.md]: can you get the other PR merged, this is not user friendly.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+With this template, we continue with our mission to make [distributed deep learning easy](https://aws.amazon.com/blogs/compute/distributed-deep-learning-made-easy/). AWS CloudFormation creates all resources in the customer account.
 
 ## What's New?  
 We've updated the AWS CloudFormation Deep Learning template to add some exciting new features and capabilities.
 
+### Nov 7 2018
+* Introduce AWS DLAMI Conda v16.0 - For developers who want pre-installed pip packages of deep learning frameworks in separate virtual environments, the Conda-based AMI is available in **[Ubuntu](https://aws.amazon.com/marketplace/pp/B077GCH38C),**  **[Amazon Linux](https://aws.amazon.com/marketplace/pp/B077GF11NF)** and **[Windows 2016](https://aws.amazon.com/marketplace/pp/B077Y5DJ8H?qid=1517601174147&sr=0-1&ref_=srh_res_product_title)** versions. Check [AWS DLAMI Official webpage](https://aws.amazon.com/machine-learning/amis/) for more details.
+* We now support 5 more AWS regions- sa-east-1, us-west-1, ca-central-1, eu-west-2, eu-west-3.
```
Comment (nswamy) [inline README.md]: can you rephrase as we support the following regions...

File: cfn-template/deeplearning.template [hunk 3]
```diff
@@ -23,8 +23,9 @@
         "p3.2xlarge",
         "p3.8xlarge",
         "p3.16xlarge",
-        "g2.8xlarge",
-        "g2.2xlarge",
```
Comment (nswamy) [inline cfn-template/deeplearning.template]: C5 instances, AMD Instances ?

Comment (mirocody) [reply cfn-template/deeplearning.template]: Updated C5, AMD instance has not been supported in DLAMI yet.

File: README.md (hunk 2 above)
Comment (mirocody) [reply README.md]: > can you rephrase as we support the following regions...

done.

**Comment by gautamkmr** [comment]
LGTM we should merge the dmlc-core PR and remove the README step from here.

File: cfn-bootstrap/dl_cfn_setup_v2.py [hunk 4]
```diff
@@ -107,6 +107,15 @@ def setup_env_variables(master_instance_ip, worker_instance_ips, default_user, e
         f.write("export DEEPLEARNING_WORKERS_PATH={}\n".format(WORKER_FILE))
         f.write("export DEEPLEARNING_WORKER_GPU_COUNT={}\n".format(gpu_count))
         f.write("export EFS_MOUNT={}\n".format(efs_mount))
+        worker_launch_template_id = os.getenv('WORKER_LAUNCH_TEMPLATE_ID')
```
[the remark this answers is not in the document]
Comment (nswamy) [reply cfn-bootstrap/dl_cfn_setup_v2.py]: It is much easier for users to say 'ssh worker1' if there is an issue than to hunt for a IP address corresponding to where the problem is

**Comment by mirocody** [comment]
Close this PR, will create a new one.

File: cfn-bootstrap/dl_cfn_setup_v2.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vikas89) [reply cfn-bootstrap/dl_cfn_setup_v2.py]: That is true, looks like new version of zmq doesn't work well with cnames.
So, we can keep worker1 in /etc/host file, but in /opt/deeplearning/workers file , ip is required.
Without this ps-lite fails to start up.

so, ssh worker1 will also work and ps-lite will also come up.

I recommend doing something like this:
  with open(HOST_FILE, 'a') as hosts, open(WORKER_FILE, 'w+') as w:
        hosts.write("{} deeplearning-master\n".format(master_instance_ip))
        worker_index=1
        for worker_ip in worker_instance_ips:
            hosts.write("{} deeplearning-worker{}\n".format(worker_ip, worker_index) )
            #w.write("deeplearning-worker{}\n".format(worker_index))
            w.write("{}\n".format(worker_ip))
            worker_index += 1
